[Fixes GEOS-10845] Update the "IntelliJ QuickStart" section on Dev documentation (#6552)
geoserver/geoserver: doc/en/developer/source/quickstart/intellij.rst

----------------------------

#. Run the IntelliJ IDE
#. From the initial panel select ``Open``.

   .. image:: img/intellij_open.png
      :width: 500

#. Navigate to the ``geoserver/src/pom.xml`` directory and click ``OK``.

   .. image:: img/intellij_import.png
      :width: 350

#. When asked click on ``Open as a Project``.

   .. image:: img/intellij_project.png
      :width: 350

> #. Optionally, depending on which platform you have, IntelliJ may ask you to ``Trust the Project``.

Reviewer: Normally we write the text in the third person, so for example:

```suggestion
#. Optionally, depending on which platform you have, IntelliJ may ask to ``Trust the Project``.
```

Unfortunately, the convention is not always used consistently (even in this document, before this PR...)

   .. image:: img/intellij_project_trust.png
      :width: 350

> #. Wait for IntelliJ to ``Sync`` the dependencies, you can follow the process from the ``Build`` tab panel on the bottom.

Reviewer: ```suggestion
#. Wait for IntelliJ to ``Sync`` the dependencies, it's possible to follow the process from the ``Build`` tab panel on the bottom.
```

   .. image:: img/intellij_sync.png
      :width: 500

Finalize the GeoServer Project configuration
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

#. Click ``File > Project Structure``.

   .. image:: img/intellij_project_structure.png
      :width: 300

#. Update the ``Name`` and select the correct ``SDK`` accordingly to your GeoServer version.

   .. image:: img/intellij_project_sdk.png
      :width: 400

#. Click ``File > Settings``.

   .. image:: img/intellij_project_settings.png
      :width: 300

#. From ``Build, Execution, Deployment > Compiler > Annotation Processors``, enable the ``Annotation processing``.

   .. image:: img/intellij_project_settings_annotation_processing.png

>       :width: 600

Reviewer: Here you say to enable annotations, but below, to disable them.... LOL
I checked, in my IDE they are disabled.

Author: @aaime moved to a dedicated Troubleshoot section

#. Click ``Build > Rebuild Project``.

   .. image:: img/intellij_project_rebuild.png
      :width: 300

Run GeoServer from IntelliJ
---------------------------

#. From the Project browser select the ``web-app`` module
#. Navigate to the ``org.geoserver.web`` package
#. Right-click the ``Start`` class and click to ``Modify Run Configuration...``

   .. image:: img/intellij_run.png
      :width: 500

> #. It is **important** that you correctly set the ``Working directory`` to ``src/web/app``. While you have the ``Edit Configurations`` dialog open, you can fine tune your launch environment (including setting a ``GEOSERVER_DATA_DIR`` or the ``jetty.port``). When you are happy with your settings, click ``OK``.

Reviewer: ```suggestion
#. It is **important** to correctly set the ``Working directory`` to ``src/web/app``. While having the ``Edit Configurations`` dialog open, fine tune the launch environment (including setting a ``GEOSERVER_DATA_DIR`` or the ``jetty.port``). When settings are satisfactory, click ``OK``.
```

   .. image:: img/intellij_run_config.png
      :width: 600

#. It's possible now to run GeoServer by selecting ``Run -> Run 'Start'``

   .. image:: img/intellij_run_start.png
      :width: 300

Troubleshooting
^^^^^^^^^^^^^^^

#. If there are errors such as "cannot find symbol class ASTAxisId", some generated code is not being included in the build.  Using ``wcs1_1`` as the working directory, run a ``mvn clean install``.
#. In the case of compiler errors like ``java.lang.NoSuchMethodError``, it might be due to ``Error Prone``. This tool is switched off by default, but sometimes it turns on after import to IntelliJ. There are two options to fix it:
    #. Go to Maven tool window and uncheck the ``errorprone`` profile, then click ``Reimport All Maven Projects``:

       .. image:: img/intellij_maven_errorprone.png
          :width: 400

>     #. To use ``errorprone``, notably to perform the QA checks, install the ``Error Prone Compiler`` plugin, restart the IDE and set ``Javac with error-prone`` as a default compiler for the project. Please note that this will slower the build.

Reviewer 2: ```suggestion
    #. To use ``errorprone``, notably to perform the QA checks, install the ``Error Prone Compiler`` plugin, restart the IDE and set ``Javac with error-prone`` as a default compiler for the project. Please note that this will slow down the build.
```
